### ISSUE TITLE
dcgan loss

### ISSUE THREAD

**Issue by eyaler** [roatienza/Deep-Learning-Experiments#10]
in
`https://github.com/roatienza/Deep-Learning-Experiments/blob/master/Experiments/Tensorflow/GAN/dcgan_mnist.py`
you compute the generator loss as:
`a_loss = self.adversarial.train_on_batch(noise, y)`
but this also trains the discriminator using only the fake samples.
shouldn't you freeze the discriminator weights for this part?

**Comment by blackeagle01** [comment]
@eyaler exactly my doubt

**Comment by elk-cloner** [comment]
> in
> `https://github.com/roatienza/Deep-Learning-Experiments/blob/master/Experiments/Tensorflow/GAN/dcgan_mnist.py`
> you compute the generator loss as:
> `a_loss = self.adversarial.train_on_batch(noise, y)`
> but this also trains the discriminator using only the fake samples.
> shouldn't you freeze the discriminator weights for this part?

you're right. we should freeze discriminator
